### PR TITLE
Concept docs for pagination guidelines

### PR DESCRIPTION
**Pull request by ikas**
Related to [GTC-406](https://gfw.atlassian.net/browse/GTC-406).

### CODE REVIEW

File: source/includes/concept/_overview.md [hunk 1]
```diff
@@ -222,4 +222,104 @@ The following endpoints adhere to the Filtering conventions defined above:
...
+curl -X GET https://api.resourcewatch.org/v1/dataset?page[number]=3
+```
+
+> Example request fetching the 5th page of 20 elements (81st - 100th elements):
+
+```shell
+curl -X GET https://api.resourcewatch.org/v1/dataset?page[number]=5&page[size]=20
+```
+
+Many times, when you're calling RW API's list endpoints, there will be a lot of results to return. Without pagination, a simple search could return hundreds or even thousands of elements, causing extraneous network traffic. For that reason, many services list their resources as pages, to make sure that not only responses are easier to handle, but also that services are scalable. Paged results have a built-in default limit of 10 elements, but we recommend you always explicitly set the limit parameter to ensure you know how many results per page you'll get.
```
Comment (tiagojsag) [inline source/includes/concept/_overview.md]: Paged results have a built-in default limit of 10 elements -> Most paged results have a built-in default limit of 10 elements

Also, not sure if "paginated" would be better, as I think that's the term we use in other places on the doc, but up to you

File: source/includes/concept/_overview.md [hunk 2]
```diff
@@ -222,4 +222,104 @@ The following endpoints adhere to the Filtering conventions defined above:
...
+Many times, when you're calling RW API's list endpoints, there will be a lot of results to return. Without pagination, a simple search could return hundreds or even thousands of elements, causing extraneous network traffic. For that reason, many services list their resources as pages, to make sure that not only responses are easier to handle, but also that services are scalable. Paged results have a built-in default limit of 10 elements, but we recommend you always explicitly set the limit parameter to ensure you know how many results per page you'll get. 
+
+The pagination strategy used across the RW API relies on two query parameters:
+
+Field          | Description                   |   Type |   Default
+-------------- | ----------------------------- | -----: | ----------:
+`page[size]`   | The number elements per page. | Number | 10
+`page[number]` | The page number.              | Number | 1
+
+Note that values above 100 for the `page[size]` query parameter are not officially supported. Also, pagination requires an implied ordering - by default, the item’s unique identifier is used, but you can use any other model field to sort the returned results.
```
Comment (tiagojsag) [inline source/includes/concept/_overview.md]: "by default, the item’s unique identifier is used" -> we make no such commitment. So we should add a few things here:
- For pagination to work reliably, users must specify a sorting criteria
- If a sorting criteria is not provided, the overall order of the elements may change between requests
- if users do not explicitly define a sorting criteria, pagination will still work, but the actual content of the pages may show missing and/or duplicated results.

Comment (tiagojsag) [reply source/includes/concept/_overview.md]: Also: may be worth adding something like "refer to [general docs on sorting] and the sorting section for the RW API resource you're loading for details on sorting option available for that resource type"

**Review by tiagojsag — CHANGES_REQUESTED** [review]
minor things here and there, overall LGTM

**Review by tiagojsag — APPROVED** [review]

File: source/includes/api/_collections.md [hunk 3]
```diff
@@ -236,22 +236,18 @@ curl -X GET https://api.resourcewatch.org/v1/collection \
...
 -H "Authorization: Bearer <your-token>"
 ```
 
-Field        |         Description          |   Type |   Default
------------- | :--------------------------: | -----: | ----------:
-page[size]   | The number elements per page.| Number | 9999999
-page[number] |       The page number        | Number | 1
+The Collections service adheres to the conventions defined in the [Pagination guidelines for the RW API](/index-rw.html#pagination), so we recommend reading that section for more details on how paginate your collections list.
+
+In the specific case of the Collections service, the default value for the `page[size]` query parameter is 9999999, instead of 10. However (as recommended in the pagination guidelines), you should not rely on the default page size and always provide a value tailored to your needs.
```
Comment (ethanroday) [inline source/includes/api/_collections.md:250]: That's a lot!

File: source/includes/api/_user_management.md [hunk 4]
```diff
@@ -470,21 +470,15 @@ Error code     | Error message  | Description
...
 ```
 
-By default, users are listed in pages of 10 results each, and the first page is loaded. However, you can customize this behavior using the following query parameters:  
-
-Field        |         Description          |   Type |   Default
------------- | :--------------------------: | -----: | ----------:
-page[size]   | The number elements per page. Values above 100 are not officially supported. | Number | 10
-page[number] |       The page number        | Number | 1
-
+The Users service adheres to the conventions defined in the [Pagination guidelines for the RW API](/index-rw.html#pagination), so we recommend reading that section for more details on how paginate your users list.
```
Comment (ethanroday) [inline source/includes/api/_user_management.md:481]: Wasn't there an issue about how pagination will be handled with the Okta API? I suppose that's not in production yet...I'll make a task to change this once it is.

**Review by ethanroday — APPROVED** [review]
LGTM!

**Pull request merged**